packages/design-system-react/src/components/button-base/ButtonBase.tsx

```diff
@@ -32,20 +32,24 @@ export const ButtonBase = React.forwardRef<HTMLButtonElement, ButtonBaseProps>(
     ref,
   ) => {
     const Component = asChild ? Slot : 'button';
+    const isInteractive = !(isDisabled ?? isLoading);
 
     const renderLoadingState = () => (
-      <span className="inline-flex items-center">
-        <Icon
-          name={IconName.Loading}
-          size={IconSize.Sm}
-          className={twMerge(
-            'animate-spin mr-2 text-inherit',
-            loadingIconProps?.className,
-          )}
-          {...loadingIconProps}
-        />
-        {loadingText ?? children}
-      </span>
+      <>
+        <span className="absolute inline-flex items-center">
+          <Icon
+            name={IconName.Loading}
+            size={IconSize.Sm}
+            className={twMerge(
+              'animate-spin mr-2 text-inherit',
+              loadingIconProps?.className,
+            )}
+            {...loadingIconProps}
+          />
+          {loadingText}
+        </span>
+        <span className="invisible inline-flex items-center">{children}</span>
+      </>
     );
```

Comment:
Refactored the `isLoading` logic to ensure the button maintains the same size as its children, regardless of the `loadingText` provided.